### PR TITLE
Ports: Add Jagged Alliance 2 Stracciatella port

### PR DESCRIPTION
**Pull request by safarp**
The port runs as demonstrated at [#ports](https://discord.com/channels/830522505605283862/830807158047244329/950382001997873182) discord channel.

There was an issue with text visualization, as you can see on the screenshot attached to the discord message (bottom right corner). The root cause was in SerenityOS not handling and printing wide characters correctly. Fixes are part of #13147 and #13151.

### CODE REVIEW

File: Ports/ja2/package.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=Jagged-Alliance-2
+version=0.15.x
+depends=("SDL2")
+workdir=ja2-stracciatella-0.15.x
+files="https://github.com/ja2-stracciatella/ja2-stracciatella/archive/refs/heads/${version}.zip ja2-stracciatella-0.x.15.zip 178375de4859d16a76276c781455bf48d3fa862841387c8aa6cfa4162f4f0ca4"
```
Comment (linusg) [inline Ports/ja2/package.sh:6]: `0.x.15`?

**Review by linusg — CHANGES_REQUESTED** [review]
We're not merging a 3.5k lines patch for this. Don't reformat code in patches, that's unacceptable.

image

File: Ports/ja2/package.sh (hunk 1 above)
Comment (safarp) [reply Ports/ja2/package.sh:6]: I must have missed reformatting in the patch, sorry for that, will fix it.
0.x.15 is a typo, will fix it as well.

Thx

**Comment by safarp** [comment]
I have removed all the reformat code from the patch and also fixed the typo in the package.sh.

**Comment by safarp** [comment]
@linusg can you please take a look again? I fixed the reformatting and also the typo.

Not really sure what you meant by 
> We're not merging a 3.5k lines patch for this. 

The patch is approx. 2.5k lines now. There is not much else I could do about it. This amount of code is needed to port project from using SDL 1.2 to 2.0. I could change it in a way that the files will be downloaded from my personal fork of ja2-stracciatella, and no patch will be needed, but I think it is better when it is downloaded from the original repository.

File: Ports/ja2/package.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=Jagged-Alliance-2
```
Comment (linusg) [inline Ports/ja2/package.sh:2]: I think this should always match the directory name:

```suggestion
port=ja2
```

File: Ports/ja2/package.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+version=0.15.x
+depends=("SDL2")
+workdir=ja2-stracciatella-0.15.x
+files="https://github.com/ja2-stracciatella/ja2-stracciatella/archive/refs/heads/${version}.zip ja2-stracciatella-0.15.x.zip 178375de4859d16a76276c781455bf48d3fa862841387c8aa6cfa4162f4f0ca4"
```
Comment (linusg) [inline Ports/ja2/package.sh:6]: Please hardcode the version in as few places as possible:

```suggestion
workdir="ja2-stracciatella-${version}"
files="https://github.com/ja2-stracciatella/ja2-stracciatella/archive/refs/heads/${version}.zip ja2-stracciatella-${version}.zip 178375de4859d16a76276c781455bf48d3fa862841387c8aa6cfa4162f4f0ca4"
```

**Review by linusg — COMMENTED** [review]
Much better, thanks! Two small suggestions :)

**Comment by safarp** [comment]
@linusg I accidentally deleted the branch used for this PR so it get Closed. I restored the branch and reopened this PR. Sorry for inconvenience. I think that once the check passes, the PR is ready to be merged. Thx

**Pull request closed without merging**